packages/suite/src/views/wallet/tokens/TokensNavigation.tsx

```diff
@@ -1,46 +1,75 @@
 import { Dispatch, SetStateAction, useEffect, useState } from 'react';
 
 import { SelectedAccountLoaded } from '@suite-common/wallet-types';
-import { selectCoinDefinitions } from '@suite-common/token-definitions';
-import { IconButton, Row, SubTabs } from '@trezor/components';
+import { selectCoinDefinitions, selectNftDefinitions } from '@suite-common/token-definitions';
+import { IconButton, IconName, Row, SubTabs } from '@trezor/components';
 import { EventType, analytics } from '@trezor/suite-analytics';
 import { Route } from '@suite-common/suite-types';
 
 import { useDispatch, useSelector } from 'src/hooks/suite';
-import { getTokens } from 'src/utils/wallet/tokenUtils';
+import { getTokens, GetTokensOutputType } from 'src/utils/wallet/tokenUtils';
 import { selectIsDebugModeActive } from 'src/reducers/suite/suiteReducer';
 import { selectRouteName } from 'src/reducers/suite/routerReducer';
 import { SearchAction } from 'src/components/wallet/SearchAction';
 import { openModal } from 'src/actions/suite/modalActions';
 import { Translation } from 'src/components/suite';
 import { goto } from 'src/actions/suite/routerActions';
 
+import { TranslationKey } from '../../../components/suite/Translation';
+
+type SubTabConfig = {
+    isNft: boolean;
+    tokens: GetTokensOutputType;
+    goToRoute: (route: Route['name']) => () => void;
+};
+
+const getSubTabConfig = ({ isNft, tokens, goToRoute }: SubTabConfig) => [
+    {
+        id: isNft ? 'wallet-nfts' : 'wallet-tokens',
+        iconName: isNft ? 'pictureFrame' : ('tokens' as IconName),
```

Comment:
This looks odd, the type cast is only applied to one part of the conditional? 
Also I recommend `satisfies` instead of `as`, because then TS actually checks the type conformity.

Reply:
That's weird indeed, good catch 

I did it this way


```
type SubTabItem = {
    id: string;
    iconName: IconName;
    onClick: () => void;
    count: number;
    labelId: TranslationKey;
};

const getSubTabConfig = ({ isNft, tokens, goToRoute }: SubTabConfig) =>
    [
        {
            id: isNft ? 'wallet-nfts' : 'wallet-tokens',
            iconName: isNft ? 'pictureFrame' : 'tokens',
            onClick: goToRoute(isNft ? 'wallet-nfts' : 'wallet-tokens'),
            count: tokens.shownWithBalance.length,
            labelId: isNft ? 'TR_NAV_COLLECTIONS' : 'TR_NAV_TOKENS',
        },
        {
            id: isNft ? 'wallet-nfts-hidden' : 'wallet-tokens-hidden',
            iconName: 'hide',
            onClick: goToRoute(isNft ? 'wallet-nfts-hidden' : 'wallet-tokens-hidden'),
            count: tokens.hiddenWithBalance.length,
            labelId: 'TR_HIDDEN',
        },
    ] satisfies SubTabItem[];
```